app/controllers/topics_controller.rb

```diff
@@ -1308,6 +1308,7 @@ def self.defer_topic_view(topic_id, ip, user_id = nil)
     Scheduler::Defer.later "Topic View" do
       topic = Topic.find_by(id: topic_id)
       next if topic.blank?
+      next if topic.shared_draft.present?
```

Comment:
Can we do an `includes(:shared_draft)` on `find_by` so we don't have to do a second trip to the DB for this?

Reply:
Actually `includes` still issues a second query. Because we're not referencing `shared_drafts` in the query it picks `#preload` as the strategy, not `#eager_load`.

It's hard to tell if two trivial queries is faster than one complex join query here, but I added a `shared_draft?` method on `Topic` that doesn't load the shared draft instead.

Reply:
Oh I had no idea there was a different strategy used at different times, I always thought `includes` would eager load the other data straight away